src/screens/UnlockScreen.tsx

```diff
@@ -44,19 +44,21 @@ export class UnlockScreen extends PureComponent<Props, State> {
   };
 
   updatePin = (pin: string) => {
-    this.setState({ pin: this.state.pin + pin }, async () => {
-      if (this.state.pin.length === CONST.pinCodeLength) {
-        const setPin = await SecureStorageService.getSecuredValue('pin');
-        if (setPin === this.state.pin) {
-          this.props.onSuccessfullyAuthenticated && this.props.onSuccessfullyAuthenticated();
-        } else {
-          this.setState({
-            error: i18n.onboarding.pinDoesNotMatch,
-            pin: '',
-          });
+    if (this.state.pin.length < CONST.pinCodeLength) {
+      this.setState({ pin: this.state.pin + pin }, async () => {
+        if (this.state.pin.length === CONST.pinCodeLength) {
+          const setPin = await SecureStorageService.getSecuredValue('pin');
```

Comment:
setPin -> pin, this is not a function to set pin but the pin itself

Reply:
Yes. It prevents state.pin to be longer than CONST.pinCodeLength and this was the case.